### PR TITLE
[lua][sql][core] DMG Limit Implementation & Fix WS pDIF cap values

### PR DESCRIPTION
**Pull request by AppleCronch**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Implements Damage Limit modifiers based on info from ffxiah and bgwiki. Increases pDIF cap based on traits and gear. ~~Implemented on core auto attack and weaponskill lua, lua implementation is on hold waiting for #5190 to merge.
https://www.bg-wiki.com/ffxi/Damage_Limit%2B https://www.ffxiah.com/forum/topic/56649/physical-damage-limit/~~

~~Traits are not implemented because the trait id is unknown, but placeholders are made with just trait id and content id needing to be filled.~~
Only content tag needs to be filled

Also changes melee WS pDIF cap from 2.25 (which I think is based on an outdated article https://ffxiclopedia.fandom.com/wiki/Level_Correction_Function_and_pDIF ) to the same values as used on auto attacks  which is based on the more updated bgwiki article and also aligns with auto attack pDIF cap.
## Steps to test these changes
Spawn some items with damage limit augments, and compare the damage values with a high attack character on a level 1 mob with and without the augmented damage limit gear. Tested on both auto attack and weaponskills for melee and range.

### CODE REVIEW

**Comment by MowFord** [comment]
https://github.com/LandSandBoat/server/blob/base/documentation/player_traits.txt for trait IDs

**Comment by AppleCronch** [comment]
Oh somehow missed it, thanks for the link!

**Comment by AppleCronch** [comment]
Updated to follow changes from #5190

File: scripts/globals/weaponskills.lua [hunk 1]
```diff
@@ -1153,6 +1157,30 @@ end
 
 -- Given the raw ratio value (atk/def) and levels, returns the cRatio (min then max)
 xi.weaponskills.cMeleeRatio = function(attacker, defender, params, ignoredDef, tp)
+    local weaponType = attacker:getWeaponSkillType(xi.slot.MAIN)
+    local pDifWeaponCapTable =
```
Comment (WinterSolstice8) [inline scripts/globals/weaponskills.lua]: this table shouldn't be copied -- you can change physical_utilities like so:

```diff
diff --git a/scripts/globals/combat/physical_utilities.lua b/scripts/globals/combat/physical_utilities.lua
index 8ad5631918..612c4aa990 100644
--- a/scripts/globals/combat/physical_utilities.lua
+++ b/scripts/globals/combat/physical_utilities.lua
@@ -46,7 +46,7 @@ local wsElementalProperties =
 }
 
 -- Table with pDIF caps per weapon/skill type.
-local pDifWeaponCapTable =
+xi.combat.physical.pDifWeaponCapTable =
 {
     -- [Skill/weapon type used] = {pre-randomizer_pDIF_cap}, Values from: https://www.bg-wiki.com/ffxi/PDIF
     [xi.skill.NONE            ] = { 3    }, -- We will use this for mobs.
@@ -384,7 +384,7 @@ xi.combat.physical.calculateMeleePDIF = function(actor, target, weaponType, wsAt
     local wRatio       = cRatio + (isCritical and 1.0 or 0)
     local pDifUpperCap = 0
     local pDifLowerCap = 0
-    local pDifFinalCap = pDifWeaponCapTable[weaponType][1] + (isCritical and 1.0 or 0) -- TODO: Add 'Damage Limit +' Trait here.
+    local pDifFinalCap = xi.combat.physical.pDifWeaponCapTable[weaponType][1] + (isCritical and 1.0 or 0) -- TODO: Add 'Damage Limit +' Trait here.
 
     -- pDIF upper cap.
     if wRatio < 0.5 then
@@ -516,7 +516,7 @@ xi.combat.physical.calculateRangedPDIF = function(actor, target, weaponType, wsA
     ----------------------------------------
     -- Step 4: Apply weapon type caps.
     ----------------------------------------
-    local pDifFinalCap = pDifWeaponCapTable[weaponType][1] -- TODO: Add 'Damage Limit +' Trait here.
+    local pDifFinalCap = xi.combat.physical.pDifWeaponCapTable[weaponType][1] -- TODO: Add 'Damage Limit +' Trait here.
 
     pDif = utils.clamp(pDif, 0, pDifFinalCap)
```

 then directly reference `xi.combat.physical.pDifWeaponCapTable`

Comment (AppleCronch) [reply scripts/globals/weaponskills.lua]: Ah, thanks for the help!

**Review by WinterSolstice8 — APPROVED** [review]
LGTM

**Review by Xaver-DaRed — APPROVED** [review]

**Pull request merged**